pycalphad/core/eqsolver.pyx

```diff
@@ -235,6 +235,22 @@ def _solve_eq_at_conditions(properties, phase_records, grid, conds_keys, state_v
             composition_sets.append(compset)
         chemical_potentials = prop_MU_values[it.multi_index]
         energy = prop_GM_values[it.multi_index]
+        # Add unrepresented phases as metastable composition sets
+        # This should help catch phases around the limit of stability
+        unrepresented_phases = sorted(set(phase_records.keys()) - set(prop_Phase_values[it.multi_index]) - {'', '_FAKE_'})
+        driving_forces = (chemical_potentials * grid.X[np.index_exp[curr_idx]]).sum(axis=-1) - grid.GM[np.index_exp[curr_idx]]
+        for phase_name in unrepresented_phases:
+            phase_record = phase_records[phase_name]
+            phase_amt = 0.0
+            compset = CompositionSet(phase_record)
+            idx_grid_for_phase = grid.Phase[np.index_exp[curr_idx]] == phase_name
+            minimum_df_idx = np.argmax(driving_forces[idx_grid_for_phase])
+            if driving_forces[idx_grid_for_phase][minimum_df_idx] < 1000:
+                continue
+            df_idx = np.arange(len(driving_forces))[idx_grid_for_phase][minimum_df_idx]
+            sfx = np.atleast_1d(np.squeeze(grid.Y[curr_idx + (df_idx,) + np.index_exp[:phase_record.phase_dof]]))
+            compset.update(sfx, phase_amt, state_variable_values)
+            composition_sets.append(compset)
```

Comment:
The changes in `eqsolver.pyx` can be reverted and worked in a future PR per offline discussion. It's more correct to write `if driving_forces[idx_grid_for_phase][minimum_df_idx] < -1000` (note the flipped sign) but the larger issue is `driving_forces` is sometimes containing positive values, which should be impossible given a well-formed starting point. That issue will be worked separately from this PR.